src/Gui/Inventor/SoAxisCrossKit.cpp

```diff
@@ -218,12 +223,28 @@ SoAxisCrossKit::createAxes()
    set("xAxis.appearance.drawStyle", "lineWidth 1");
    set("yAxis.appearance.drawStyle", "lineWidth 1");
    set("zAxis.appearance.drawStyle", "lineWidth 1");
-   set("xAxis.appearance.material", "diffuseColor 0.5 0.125 0.125");
-   set("xHead.appearance.material", "diffuseColor 0.5 0.125 0.125");
-   set("yAxis.appearance.material", "diffuseColor 0.125 0.5 0.125");
-   set("yHead.appearance.material", "diffuseColor 0.125 0.5 0.125");
-   set("zAxis.appearance.material", "diffuseColor 0.125 0.125 0.5");
-   set("zHead.appearance.material", "diffuseColor 0.125 0.125 0.5");
+
+   unsigned long colorLong;
+   App::Color color;
+   std::stringstream parameterstring;
+
+   colorLong = Gui::ViewParams::instance()->getAxisXColor();
+   color = App::Color(static_cast<uint32_t>(colorLong));
```

Comment:
This conversion is done very often, `getAxisXColor` should probably just return `App::Color`. 

I feel that current version could be renamed to `getAxisXPackedColor`, but maybe it won't be needed as I bet that most uses convert it anyway.

Reply:
```
getAxisXColor<uint32_t>()
getAxisXColor<App::Color>()
```